kernel/src/process.rs

```diff
@@ -1473,16 +1473,65 @@ impl<C: Chip> ProcessType for Process<'_, C> {
             let _ = writer.write_fmt(format_args!("{}", config));
         });
 
-        let sram_start = self.memory.as_ptr() as usize;
-        let flash_start = self.flash.as_ptr() as usize;
-        let flash_init_fn = flash_start + self.header.get_init_function_offset() as usize;
+        /// Return Ok(Flash Address, RAM address) if this app has
+        /// fixed_addresses, Err(()) otherwise.
+        fn check_for_fixed_address(flash: &'static [u8]) -> Result<(u32, u32), ()> {
```

Comment:
Why make this a nested function if you're only going to call it once?

Reply:
So I can use the `?` operator. I'm not sure how else to do it.